deeplake/core/dataset/dataset.py

```diff
@@ -3244,6 +3248,58 @@ def _delete_metas(self):
             except KeyError:
                 pass
 
+    def connect(
+        self,
+        creds_key: str,
+        dest_path: Optional[str] = None,
+        org_id: Optional[str] = None,
+        ds_name: Optional[str] = None,
+        token: Optional[str] = None,
+    ):
+        """Connect a Deep Lake cloud dataset through a deeplake path.
+
+        Examples:
+            >>> # create/load an s3 dataset
+            >>> s3_ds = deeplake.dataset("s3://bucket/dataset")
+            >>> ds = s3_ds.connect(dest_path="hub://my_org/dataset", creds_key="my_managed_credentials_key")
+            >>> # or
+            >>> ds = s3_ds.connect(org_id="my_org", creds_key="my_managed_credentials_key")
+
+        Args:
+            creds_key (str): The managed credentials to be used for accessing the source path.
+            dest_path (str, optional): The full path to where the connected Deep Lake dataset will reside. Can be:
+                a Deep Lake path like ``hub://organization/dataset``
+            org_id (str, optional): The organization to where the connected Deep Lake dataset will be added.
+            ds_name (str, optional): The name of the connected Deep Lake dataset. Will be infered from ``dest_path`` or ``src_path`` if not provided.
+            token (str, optional): Activeloop token used to fetch the managed credentials.
+
+        Returns:
+            Dataset: The connected Deep Lake dataset.
+
+        Raises:
+            InvalidSourcePathError: If the dataset's path is not a valid s3 or gcs path.
+            InvalidDestinationPathError: If ``dest_path``, or ``org_id`` and ``ds_name`` do not form a valid Deep Lake path.
+        """
+        connected_id = connect_dataset_entry(
```

Comment:
this is almost the same code as api/dataset. Can you refactor this out to a function that can be called in both places?

Reply:
Actually now that the DatasetEntry handles most of the stuff, I logging and reporting to the util function